lib/toolkit/loader.js

```diff
@@ -623,17 +625,26 @@ const Require = iced(function Require(loader, requirer) {
       throw Error('you must provide a module name when calling require() from '
                   + requirer.id, requirer.uri);
 
-    let requirement;
-    let uri;
+    let requirement, uri;
 
     // TODO should get native Firefox modules before doing node-style lookups
     // to save on loading time
     if (isNative) {
       // If a requireMap is available from `generateMap`, use that to
       // immediately resolve the node-style mapping.
+      // TODO: write more tests for this use case
       if (requireMap && requireMap[requirer.id])
         requirement = requireMap[requirer.id][id];
 
+      let { overrides } = manifest.jetpack;
+      if (overrides) {
+        for (let key in overrides) {
+          if (id == key || (id.substr(0, key.length + 1) == (key + "/"))) {
```

Comment:
Is this statement checking for trailing slashes? Should add a comment, a lot of this require code is hard to read (partially my fault) for lack of comments and scenarios